Description: Ubuntu ships compatibility names for .desktop that got renamed
 upstream, those use NoDisplay but the panel doesn't respect that key,
 skip known buggy entries for file-roller and nautilus
Origin: ubuntu
Bug-Ubuntu: https://launchpad.net/bugs/1716267
Forwarded: not-needed
diff --git a/panels/notifications/cc-notifications-panel.c b/panels/notifications/cc-notifications-panel.c
index 89e33e171..4d470e7c0 100644
--- a/panels/notifications/cc-notifications-panel.c
+++ b/panels/notifications/cc-notifications-panel.c
@@ -470,6 +470,14 @@ process_app_info (CcNotificationsPanel *panel,
   guint i;
 
   app_id = app_info_get_id (app_info);
+
+  /* Ignore compatibility desktops (lp: #1716267) */
+  if (g_strcmp0 (app_id, "file-roller") == 0)
+    return;
+
+  if (g_strcmp0 (app_id, "nautilus") == 0)
+    return;
+
   canonical_app_id = g_strcanon (app_id,
                                  "0123456789"
                                  "abcdefghijklmnopqrstuvwxyz"
